src/components/UndoButton/UndoButtonAdd.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '@patternfly/react-core';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { FormattedMessage } from 'react-intl';
+
+import { removeMessage, undoAddSource } from '../../redux/actions/providers';
+import { paths } from '../../Routes';
+import { refreshPage } from './refreshPage';
+
+const UndoButton = ({
+    notifications,
+    messageId,
+    history,
+    removeMessage,
+    values,
+    undoAddSource
+}) => (
+    <Button variant="link" isInline onClick={() => {
+        const notification = notifications.find(({ customId }) => customId === messageId);
+
+        if (notification) {
+            removeMessage(notification.id);
```

Comment:
So the notifications middleware allows clearing all notifications and there is specific action for it. Maybe you could do that instead of finding the notification? Don't know if closing all is useful here

Reply:
User can have a stack of notifications.

(And it allows to stack multiple undoes :D )